### PR TITLE
connect-replication-channel-to-dht

### LINKED ISSUE #59: Connect republish/replication channel to dht

### PR DESCRIPTION
**Pull request by edvinnn**
Fixes #59

### CODE REVIEW

File: dht/handler.go [hunk 1]
```diff
@@ -99,3 +99,11 @@ func (dht *DHT) pongRequestHandler() {
 		}
 	}
 }
+
+func (dht *DHT) republishRequestHandler() {
+	for {
+		request := <- dht.nw.RepublishRequestCH()
```
Comment (optmzr) [inline dht/handler.go]: You want to use this method instead:
```go
dht.db.LocalItemCh()
```

And then do a:
```go
dht.Put(value)
```

With the value from the channel.

Comment (Hjortsberg) [reply dht/handler.go]: Correct! :smile:

**Comment by Hjortsberg** [comment]
Just realized I named the channel badly. If a replication event occurs, it will publish remoteValues to the channel, I.E not only localItems as the name suggests.

**Comment by optmzr** [comment]
Yeah, but the function will be the same. So a "ItemCh" is a good enough name?

**Comment by optmzr** [comment]
So Edvin could just rename it :)

**Comment by Hjortsberg** [comment]
Yes. :+1:

File: dht/handler.go [hunk 2]
```diff
@@ -99,3 +99,20 @@ func (dht *DHT) pongRequestHandler() {
 		}
 	}
 }
+
+func (dht *DHT) republishRequestHandler() {
+	for {
+		value := <- dht.db.LocalItemCh()
+
+		go func(val string) {
```
Comment (optmzr) [inline dht/handler.go]: I don't think we want to do this in a go routine, as we want to be somewhat near the alpha-parameter?

Comment (edvinnn) [reply dht/handler.go]: Ok

**Pull request merged**